tests/test_view.py

```diff
@@ -111,190 +103,231 @@ def normalize(self, vec):
-        initial_up_vector = self.up_vector()
+        c = self.camera
 
-        self.move(x=1, y=1, z=1, dt=dt)
-        self.move(yaw=1, pitch=1, dt=dt, left_mouse_button=True)
+        initial_position = c.position
+        initial_view_center = c.view_center
+        initial_up_vector = c.up_vector
 
-        self.assertNotEqual(self.pos(), initial_pos)
-        self.assertNotEqual(self.view_vector(), initial_view_vector)
-        self.assertNotEqual(self.view_center(), initial_view_center)
-        self.assertNotEqual(self.up_vector(), initial_up_vector)
+        c.move(x=0.1, y=0.1, z=0.1)
+        c.move(yaw=1, pitch=1, left_mouse_button=True)
 
-        self.widget.resetCamera()
+        self.assertNotEqual(c.position, initial_position)
+        self.assertNotEqual(c.view_center, initial_view_center)
+        self.assertNotEqual(c.up_vector, initial_up_vector)
 
-        self.assertEqual(self.pos(), initial_pos)
-        self.assertEqual(self.view_vector(), initial_view_vector)
-        self.assertEqual(self.view_center(), initial_view_center)
-        self.assertEqual(self.up_vector(), initial_up_vector)
+        c.reset()
+
+        self.assertEqual(c.position, initial_position)
+        self.assertEqual(c.view_center, initial_view_center)
+        self.assertEqual(c.up_vector, initial_up_vector)
 
     def test_translation(self):
-        dt = 0.01
-        delta = dt * self.linear_speed()
-        delta_vec = np.array([delta, delta, -delta])
-        new_pos = np.array(self.pos()) + delta_vec
-        new_view_center = np.array(self.view_center()) + delta_vec
+        c = self.camera
+
+        speed = c.linear_speed
+        delta_vec = np.array([speed, speed, -speed])
+        new_position = np.array(c.position) + delta_vec
+        new_view_center = np.array(c.view_center) + delta_vec
 
-        self.move(x=1, dt=dt)
-        self.assertEqual(self.pos()[0], new_pos[0])
+        c.move(x=1)
+        self.assertEqual(c.position[0], new_position[0])
 
-        self.move(y=1, dt=dt)
-        self.assertEqual(self.pos()[1], new_pos[1])
+        c.move(y=1)
+        self.assertEqual(c.position[1], new_position[1])
 
         # camera moves in negative z direction
-        self.move(z=1, dt=dt)
-        self.assertEqual(self.pos()[2], new_pos[2])
+        c.move(z=1)
+        self.assertEqual(c.position[2], new_position[2])
 
         # camera view center should move with camera
-        view_center = self.view_center()
-        self.assertEqual(view_center[0], new_view_center[0])
-        self.assertEqual(view_center[1], new_view_center[1])
-        self.assertEqual(view_center[2], new_view_center[2])
+        self.assertEqual(c.view_center[0], new_view_center[0])
+        self.assertEqual(c.view_center[1], new_view_center[1])
+        self.assertEqual(c.view_center[2], new_view_center[2])
 
     def test_rotation(self):
-        dt = 0.01
-        angle = dt * self.look_speed()
+        c = self.camera
```

Comment:
Personally, I don't like a shorthand like `c`, but maybe yyc has a different thought.

Reply:
I did that due to flake8's rule to limit lines to 80 characters. If I write `self.camera` or `camera` everywhere I will get a lot of errors and I will have to shorten more other variables :/

Reply:
A local `c` in test code is good.  It can't be right to skim through testing code for a complex structure for 3D camera with only a glance.  Writing `c = self.camera` forces a maintainer to pause for a second to keep the information in their brain's short-term memory block, and improves their understanding of the code.